lbrynet/extras/daemon/Daemon.py

```diff
@@ -3328,173 +3328,122 @@ def jsonrpc_routing_table_get(self):
+            --channel_id=<channel_id>   : (str) The ID of the channel you want to post under
+            --channel_name=<channel_name>   : (str) The channel you want to post as, prepend with a '@'
+
 
         Returns:
-            (dict) Comment object if successfully made
-        """
-        if not 1 < len(message) <= 2000:
-            raise Exception(f'Message length ({len(message)}) needs to be between 2 and 2000 chars')
-        url = self.conf.comment_server
-        if parent_comment_id is not None:
-            comment_id = await jsonrpc_post(url, 'reply', parent_id=parent_comment_id,
-                                            poster=channel_id, message=message)
-        else:
-            claim_data = await self.jsonrpc_claim_search(claim_id=claim_id)
-            if 'error' not in claim_data and claim_data['total_pages'] == 1:
-                uri = claim_data['items'][0].permanent_url
-                comment_id = await jsonrpc_post(url, 'comment', uri=uri,
-                                                poster=channel_id, message=message)
-            else:
-                raise Exception(f"permanent_url is not in the claim_data {claim_data}\n"
-                                f"The given claim_id ({claim_id}) may be invalid")
-        return await jsonrpc_post(url, 'get_comment_data', comm_index=comment_id, better_keys=True)
+            (dict) Comment object if successfully made, (None) otherwise
+            {
+                "comment":      (str) The actual string as inputted by the user,
+                "comment_id":   (str) The Comment's unique identifier,
+                "channel_name": (str) Name of the channel this was posted under, prepended with a '@',
+                "channel_id":   (str) The Channel Claim ID that this comeent was posted under,
+                "signature":    (str) The signature of the comment,
+                "channel_url":  (str) Channel's URI in the ClaimTrie,
+                "parent_id":    (str) Comment this is replying to, (None) if this is the root,
+                "timestamp":    (int) The time at which comment was entered into the server at, in nanoseconds.
+            }
+        """
+        if bool(channel_name) ^ bool(channel_id):
+            try:
+                channel_list = await self.jsonrpc_channel_list()
+                if channel_name:
+                    channel_name = channel_name.lower()
+                    channel: Output = [chan for chan in channel_list if chan.normalized_name == channel_name].pop()
+                    channel_id = channel.claim_id
+                else:
+                    channel: Output = [chan for chan in channel_list if chan.claim_id == channel_id].pop()
+                    channel_name = channel.normalized_name
+            except IndexError:
+                raise ValueError('You must enter a valid channel_%s' % ('id' if channel_id else 'name'))
```

Comment:
All of the code above is unnecessary. Take a look at how a bunch of the other commands which accept a `channel_id`/`channel_name` work.